src/garage/tf/algos/dqn.py

```diff
@@ -191,32 +191,28 @@ def train_once(self, itr, paths):
 
         """
         paths = self.process_samples(itr, paths)
-        epoch = itr / self.steps_per_epoch
+        epoch = itr // self.steps_per_epoch
 
         self.episode_rewards.extend(paths['undiscounted_returns'])
         last_average_return = np.mean(self.episode_rewards)
         for _ in range(self.n_train_steps):
             if (self.replay_buffer.n_transitions_stored >=
                     self.min_buffer_size):
-                self.evaluate = True
                 qf_loss = self.optimize_policy(epoch, None)
                 self.episode_qf_losses.append(qf_loss)
 
-        if self.evaluate:
```

Comment:
and this

Reply:
FYI: ddpg uses soft-update so it updates QFunction every iterations after `min_buffer_size.
https://github.com/rlworkgroup/garage/blob/779f87fc5bdab93b45cdd4a4d0d5e2b6949e6cda/src/garage/tf/algos/ddpg.py#L272
However, dqn uses hard-update it only updates once, hence an extra guard is required.